src/ResourceManager/DataFactoryV2/Commands.DataFactoryV2/IntegrationRuntimes/SetAzureDataFactoryIntegrationRuntimeCommand.cs

```diff
@@ -129,29 +318,90 @@ public class SetAzureDataFactoryIntegrationRuntimeCommand : IntegrationRuntimeCm
         public string LicenseType { get; set; }
 
         [Parameter(
+            ParameterSetName = ParameterSetNames.ByIntegrationRuntimeName,
+            Mandatory = false,
+            HelpMessage = Constants.HelpIntegrationRuntimeAuthKey)]
+        [Parameter(
+            ParameterSetName = ParameterSetNames.ByResourceId,
+            Mandatory = false,
+            HelpMessage = Constants.HelpIntegrationRuntimeAuthKey)]
+        [Parameter(
+            ParameterSetName = ParameterSetNames.ByIntegrationRuntimeObject,
             Mandatory = false,
             HelpMessage = Constants.HelpIntegrationRuntimeAuthKey)]
         [ValidateNotNull]
         public System.Security.SecureString AuthKey { get; set; }
 
+        [Parameter(
+            ParameterSetName = ParameterSetNames.ByLinkedIntegrationRuntimeName,
+            Mandatory = true,
+            HelpMessage = Constants.HelpSharedIntegrationRuntimeResourceId)]
+        [Parameter(
+            ParameterSetName = ParameterSetNames.ByLinkedIntegrationRuntimeResourceId,
+            Mandatory = true,
+            HelpMessage = Constants.HelpSharedIntegrationRuntimeResourceId)]
+        [Parameter(
+            ParameterSetName = ParameterSetNames.ByLinkedIntegrationRuntimeObject,
+            Mandatory = true,
+            HelpMessage = Constants.HelpSharedIntegrationRuntimeResourceId)]
+        [ValidateNotNullOrEmpty]
+        public string SharedIntegrationRuntimeResourceId { get; set; }
+
         [Parameter(
             Mandatory = false, HelpMessage = Constants.HelpDontAskConfirmation)]
         public SwitchParameter Force { get; set; }
 
+        protected override void ByResourceId()
+        {
+            if (!string.IsNullOrWhiteSpace(ResourceId))
+            {
+                var parsedResourceId = new ResourceIdentifier(ResourceId);
+                ResourceGroupName = parsedResourceId.ResourceGroupName;
+
+                var parentResource = parsedResourceId.ParentResource.Split(new[] { '/' });
+                DataFactoryName = parentResource[parentResource.Length - 1];
+
+                Name = parsedResourceId.ResourceName;
+            }
+        }
+
+        protected override void ByIntegrationRuntimeObject()
+        {
+            if (InputObject != null)
+            {
+                ResourceGroupName = InputObject.ResourceGroupName;
+                DataFactoryName = InputObject.DataFactoryName;
+                Name = InputObject.Name;
+            }
+        }
+
+
         [EnvironmentPermission(SecurityAction.Demand, Unrestricted = true)]
         public override void ExecuteCmdlet()
         {
             this.ByResourceId();
             this.ByIntegrationRuntimeObject();
 
+            if (Type.Equals(Constants.IntegrationRuntimeTypeManaged, StringComparison.OrdinalIgnoreCase))
+            {
+                if (AuthKey != null || !string.IsNullOrWhiteSpace(SharedIntegrationRuntimeResourceId))
```

Comment:
This is no longer possible because of parameter set validation, so you can remove this whole if statement.

Reply:
Ah, you're right.  Disregard my comment.